src/Validate/StatisticsGrid.php

```diff
@@ -21,9 +21,17 @@ class StatisticsGrid {
    *   The complete form structure.
    */
   public static function validate(array &$element, FormStateInterface $formState, array &$form) {
+    $deltas = $element['subform']['field_statistic_block']['widget'];
+
+    $count = 0;
+    foreach (array_keys($deltas) as $key) {
+      if (is_numeric($key)) {
+        $count += 1;
+      }
+    }
+
     $error = FALSE;
-    $max_delta = $element['subform']['field_statistic_block']['widget']['#max_delta'];
-    if ($max_delta == 0) {
+    if ($count < 2) {
       $error = TRUE;
```

Comment:
Soooo do we have a unit test for this? This seems like a really easy thing to pass some input into and validate that your count works :)

Reply:
Ed's has left , I will refactor the code and add the unit test.